{
  "uid" : "a81205df6540b394",
  "name" : "Check args is not presence.",
  "fullName" : "tests.test_cards.TestPrepositions_1_7#test_args_is_not_presence",
  "historyId" : "dbf888370986eb8b8a9dc1de6e1f8bf7",
  "time" : {
    "start" : 1698687098839,
    "stop" : 1698687098839,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Chrome failed to start: exited normally.\n  (session not created: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x55740e597fb3 <unknown>\n#1 0x55740e26b4a7 <unknown>\n#2 0x55740e29ec93 <unknown>\n#3 0x55740e29b10c <unknown>\n#4 0x55740e2ddac6 <unknown>\n#5 0x55740e2d4713 <unknown>\n#6 0x55740e2a718b <unknown>\n#7 0x55740e2a7f7e <unknown>\n#8 0x55740e55d8d8 <unknown>\n#9 0x55740e561800 <unknown>\n#10 0x55740e56bcfc <unknown>\n#11 0x55740e562418 <unknown>\n#12 0x55740e52f42f <unknown>\n#13 0x55740e5864e8 <unknown>\n#14 0x55740e5866b4 <unknown>\n#15 0x55740e597143 <unknown>\n#16 0x7f4499694ac3 <unknown>",
  "statusTrace" : "@pytest.fixture()\n    def driver():\n        options = Options()\n        # options.add_argument('--headless')\n>       driver = webdriver.Chrome(options=options)\n\ntests/conftest.py:14: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/chrome/webdriver.py:45: in __init__\n    super().__init__(\n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/chromium/webdriver.py:56: in __init__\n    super().__init__(\n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:205: in __init__\n    self.start_session(capabilities)\n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:289: in start_session\n    response = self.execute(Command.NEW_SESSION, caps)[\"value\"]\n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:344: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7faaaf624610>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"session not created\",\"message\":\"session not created: Chrome failed to sta...\\\\n#14 0x55740e5866b4 \\\\u003Cunknown>\\\\n#15 0x55740e597143 \\\\u003Cunknown>\\\\n#16 0x7f4499694ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Chrome failed to start: exited normally.\nE         (session not created: DevToolsActivePort file doesn't exist)\nE         (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nE       Stacktrace:\nE       #0 0x55740e597fb3 <unknown>\nE       #1 0x55740e26b4a7 <unknown>\nE       #2 0x55740e29ec93 <unknown>\nE       #3 0x55740e29b10c <unknown>\nE       #4 0x55740e2ddac6 <unknown>\nE       #5 0x55740e2d4713 <unknown>\nE       #6 0x55740e2a718b <unknown>\nE       #7 0x55740e2a7f7e <unknown>\nE       #8 0x55740e55d8d8 <unknown>\nE       #9 0x55740e561800 <unknown>\nE       #10 0x55740e56bcfc <unknown>\nE       #11 0x55740e562418 <unknown>\nE       #12 0x55740e52f42f <unknown>\nE       #13 0x55740e5864e8 <unknown>\nE       #14 0x55740e5866b4 <unknown>\nE       #15 0x55740e597143 <unknown>\nE       #16 0x7f4499694ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:229: SessionNotCreatedException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1698687082556,
      "stop" : 1698687082642,
      "duration" : 86
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "stepsCount" : 0,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1698687098839,
      "stop" : 1698687099575,
      "duration" : 736
    },
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Chrome failed to start: exited normally.\n  (session not created: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x55740e597fb3 <unknown>\n#1 0x55740e26b4a7 <unknown>\n#2 0x55740e29ec93 <unknown>\n#3 0x55740e29b10c <unknown>\n#4 0x55740e2ddac6 <unknown>\n#5 0x55740e2d4713 <unknown>\n#6 0x55740e2a718b <unknown>\n#7 0x55740e2a7f7e <unknown>\n#8 0x55740e55d8d8 <unknown>\n#9 0x55740e561800 <unknown>\n#10 0x55740e56bcfc <unknown>\n#11 0x55740e562418 <unknown>\n#12 0x55740e52f42f <unknown>\n#13 0x55740e5864e8 <unknown>\n#14 0x55740e5866b4 <unknown>\n#15 0x55740e597143 <unknown>\n#16 0x7f4499694ac3 <unknown>\n\n",
    "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1123, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/_pytest/fixtures.py\", line 895, in call_fixture_func\n    fixture_result = next(generator)\n  File \"/home/runner/work/English-exercises-AQA/English-exercises-AQA/tests/conftest.py\", line 14, in driver\n    driver = webdriver.Chrome(options=options)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/chrome/webdriver.py\", line 45, in __init__\n    super().__init__(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/chromium/webdriver.py\", line 56, in __init__\n    super().__init__(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 205, in __init__\n    self.start_session(capabilities)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 289, in start_session\n    response = self.execute(Command.NEW_SESSION, caps)[\"value\"]\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "stepsCount" : 0,
    "hasContent" : true
  } ],
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Prepositions with nouns."
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_cards"
  }, {
    "name" : "subSuite",
    "value" : "TestPrepositions_1_7"
  }, {
    "name" : "host",
    "value" : "fv-az414-664"
  }, {
    "name" : "thread",
    "value" : "1830-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_cards"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 2,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 2
      },
      "items" : [ {
        "uid" : "ec1a99db0da65d09",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Chrome failed to start: exited normally.\n  (session not created: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x56476b9a5fb3 <unknown>\n#1 0x56476b6794a7 <unknown>\n#2 0x56476b6acc93 <unknown>\n#3 0x56476b6a910c <unknown>\n#4 0x56476b6ebac6 <unknown>\n#5 0x56476b6e2713 <unknown>\n#6 0x56476b6b518b <unknown>\n#7 0x56476b6b5f7e <unknown>\n#8 0x56476b96b8d8 <unknown>\n#9 0x56476b96f800 <unknown>\n#10 0x56476b979cfc <unknown>\n#11 0x56476b970418 <unknown>\n#12 0x56476b93d42f <unknown>\n#13 0x56476b9944e8 <unknown>\n#14 0x56476b9946b4 <unknown>\n#15 0x56476b9a5143 <unknown>\n#16 0x7f90c5e94ac3 <unknown>",
        "time" : {
          "start" : 1698686838060,
          "stop" : 1698686838060,
          "duration" : 0
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "a81205df6540b394.json",
  "parameterValues" : [ ]
}